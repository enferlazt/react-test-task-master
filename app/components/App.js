import '../styles/application.scss';
import React, {PureComponent} from 'react';
import TickerCard from './ticker-card/TickerCard';
import InputTickerSymbol from './input-ticker-symbol/InputTickerSymbol';
import { Route, Switch } from 'react-router';
import NoContent from './no-content/NoContent';

// The below line is here as an example of getting prices
// connect('AAPL');

class App extends PureComponent {
    render() {
        return (
            <main className="stock-ticker">
                <h1 className="stock-ticker__title">Stock Blotter</h1>
                <InputTickerSymbol/>
                <Switch>
                    <Route path="/:symbol" children={<TickerCard/>} />
                    <Route path="/" exact children={<NoContent/>} />
                </Switch>
            </main>
        );
    }
}

export default App;
