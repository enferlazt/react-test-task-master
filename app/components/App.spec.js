import { shallow } from 'enzyme';
import React from 'react';
import App from './App';
import '../../setupTests';

describe("App component", () => {
    it("render component", () => {
        const component = shallow(<App/>);
        expect(component).toMatchSnapshot();
    });
});
