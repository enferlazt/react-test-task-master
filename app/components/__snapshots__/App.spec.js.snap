// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render component 1`] = `ShallowWrapper {}`;
